### PR TITLE
[1880] fix scr qsr placement blocking

### LINKED ISSUE #8853: [1880] Token Placement allowed on D12 (Shenyang & Fushun - QSR Home) before QSR started
GameID: https://18xx.games/game/111286

Specific Action Link: https://18xx.games/game/111286?action=379

JHA was able to place a token on D12, which is QSR's home. QSR has not yet started. 1880 rules prohibit placing a token on either the QSR or SCR home until the associated company's home token has been placed (see bottom of page 13 of rules pdf linked in the game Info section).

### PR DESCRIPTION
**Pull request by bentziaxl**
fix #8853

### CODE REVIEW

**Review by michaeljb — APPROVED** [review]

**Comment by bentziaxl** [comment]
Games where another corp tokened a yellow/green tile of these two starting cities will break. However, maybe they should be broken and not pinned?

File: lib/engine/tile.rb [hunk 1]
```diff
@@ -374,7 +374,9 @@ def add_reservation!(entity, city, slot = nil, reserve_city = true)
     def token_blocked_by_reservation?(corporation)
       return false if @reservations.empty?
 
-      if @reservation_blocks == :always || (@reservation_blocks == :yellow_only && @color == :yellow)
+      if @reservation_blocks == :always ||
+        (@reservation_blocks == :yellow_only && @color == :yellow) ||
+        (@reservation_blocks == :oo_only && @cities.size > 1)
```
Comment (crericha) [inline lib/engine/tile.rb:379]: This isn't correct behavior. Why was this approved and merged?

Comment (tobymao) [reply lib/engine/tile.rb:379]: lack of familiarity probably. feel free to revert

Comment (bentziaxl) [reply lib/engine/tile.rb:379]: why is it wrong?

Comment (crericha) [reply lib/engine/tile.rb:379]: @bentziaxl, the logic happens to work for 1880, but makes bad assumptions from a general reusability perspective . I provide a brief rationale here: #8879. Ping me on slack if you want to talk about it more.

**Pull request merged**